node/src/components/consensus/era_supervisor.rs

```diff
@@ -595,6 +599,102 @@ where
     }
 }
 
+/// Returns an era ID in which the booking block for `era_id` lives in, if we can use it.
+/// Booking block for era N is the switch block (the last block) in era N – AUCTION_DELAY - 1.
+/// To find it, we get the start height of era N - AUCTION_DELAY and subtract 1.
+/// We make sure not to use an era ID below the last upgrade activation point, because we will
+/// not have instances of eras from before that.
+///
+/// We can't use it if it is:
+/// * before Genesis
+/// * before upgrade
+/// * before emergency restart
+/// In those cases, returns `None`.
+fn valid_booking_block_era_id(
+    era_id: EraId,
+    auction_delay: u64,
+    last_activation_point: EraId,
+) -> Option<EraId> {
+    // If this is the first era after the upgrade/restart/Genesis,
+    // booking block is not defined.
+    if era_id == last_activation_point {
+        return None;
+    }
+
+    let booking_era_id = era_id
+        .saturating_sub(auction_delay)
+        .saturating_sub(1)
+        .max(last_activation_point);
```

Comment:
```suggestion
    let booking_era_id = era_id
        .saturating_sub(auction_delay)
        .saturating_sub(1);
```
No need to `max`, if we return `None` in those cases anyway.

If we `max` here, the `if` condition below can never be satisfied.

Reply:
Done in [`0fb82b1` (#1068)](https://github.com/CasperLabs/casper-node/pull/1068/commits/0fb82b173e1fea440596ef4c214de1c4100a5d7a)